From 8590cdc13f8bd2abc354425e03368e2f435a7e4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Le=20Goffic?= <clement.legoffic@foss.st.com>
Date: Fri, 13 Sep 2024 14:47:34 +0200
Subject: [PATCH 1153/1695] dts: stm32: add reset property for IWDG2
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Through the reset line added in this patch we are now able to stop the
IWDG2 and its 'kernel reset' behavior.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5b7662ce904fceeab6c9dbdbe1bf36d9a21b3035 dts: stm32: add reset property for IWDG2]
Change-Id: I4a57a3ae75ac229c16004a116478a77b34d54fe7
Signed-off-by: Clément Le Goffic <clement.legoffic@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/381314
Reviewed-by: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Antonio Maria BORNEO <antonio.borneo@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 1 +
 core/arch/arm/dts/stm32mp231.dtsi | 1 +
 core/arch/arm/dts/stm32mp251.dtsi | 1 +
 3 files changed, 3 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 748c74d02..5e1db084c 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -479,6 +479,7 @@
 				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
+				resets = <&rcc IWDG2_SYS_R>;
 				access-controllers = <&rifsc STM32MP21_RIFSC_IWDG2_ID>;
 				status = "disabled";
 			};
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index a03958d0b..badbebbd3 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -564,6 +564,7 @@
 				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
+				resets = <&rcc IWDG2_SYS_R>;
 				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG2_ID>;
 				status = "disabled";
 			};
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 4cfb9c7ad..313783381 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -626,6 +626,7 @@
 				interrupts = <GIC_SPI 3 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&rcc CK_BUS_IWDG2>, <&rcc LSI_CK>;
 				clock-names = "pclk", "lsi";
+				resets = <&rcc IWDG2_SYS_R>;
 				access-controllers = <&rifsc STM32MP25_RIFSC_IWDG2_ID>;
 				status = "disabled";
 			};
-- 
2.47.2

